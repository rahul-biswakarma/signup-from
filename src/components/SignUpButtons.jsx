import React from "react";

import InfoIcon from "../assets/icons/info-icon.svg";
import EmailIcon from "../assets/icons/email-icon.svg";
import GoogleIcon from "../assets/icons/google-icon.svg";

export default function SignUpButtons() {
	return (
		<div className="flex flex-col gap-[1rem] w-full max-[400px]:mt-[2rem] mt-[4rem]">
			<button
				data-testid="continue-with-google-button"
				className="relative w-full bg-blue_1 p-[0.5rem_1rem] rounded flex flex-wrap items-center"
			>
				<img
					src={GoogleIcon}
					alt="Google Icon"
					className="w-6 h-6"
				/>
				<span className="absolute text-center w-full text-white text-sm font-[500] ml-2">
					Continue with Google
				</span>
			</button>
			<a href="/signup">
				<button
					data-testid="continue-with-email-button"
					className="relative w-full bg-dark_gray p-[0.5rem_1rem] rounded flex items-center flex-wrap"
				>
					<img
						src={EmailIcon}
						alt="Google Icon"
						className="w-6 h-6"
					/>
					<span className="absolute text-center w-full text-white text-sm font-[500] ml-2">
						Continue with Email
					</span>
				</button>
			</a>
			<div className="flex items-center gap-[10px]">
				<img
					src={InfoIcon}
					alt="Info Icon"
					className="w-5 h-5"
				/>
				<p className="text-sm text-black/40">
					We recommend using your work email
				</p>
			</div>
		</div>
	);
}
