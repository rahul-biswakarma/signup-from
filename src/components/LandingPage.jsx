import { useEffect } from "react";

import Header from "./Header";
import SignUpButtons from "./SignUpButtons";

import ProductImage from "../assets/images/product-image.png";

export default function LandingPage() {
	useEffect(() => {
		document.title = "Sign into Devrev";
	}, []);

	return (
		<div
			data-testid="landing-component"
			className="w-full h-[100vh] overflow-hidden grid max-[1100px]:grid-cols-[100%] grid-cols-[700px_auto] "
		>
			<section className="flex flex-col items-center justify-center bg-bg_1 border-r-[1px] border-border_color w-full h-full p-24 max-[600px]:p-12">
				<main className="max-w-[500px]">
					<Header />
					<SignUpButtons />
					<p className="text-sm text-black/40 max-[400px]:my-[2rem] my-[4rem]">
						By continuing, you acknowledge that you have read and understood,
						and agree to DevRev's{" "}
						<strong className="text-black font-semibold">
							Terms & Conditions
						</strong>{" "}
						and{" "}
						<strong className="text-black font-semibold">Privacy Policy</strong>
						.
					</p>
					<p className="text-sm text-black/40">
						Already have an account?{" "}
						<span className="text-black font-semibold cursor-pointer">
							<a href="/login">Login</a>
						</span>
					</p>
				</main>
			</section>
			<section className="bg-bg_2 w-full h-full relative p-24 max-[1100px]:hidden">
				<h2 className="text-black max-[1200px]:text-3xl text-5xl max-w-[600px] absolute top-[5rem] max-[1300px]:left-[4rem] left-[8rem]  font-bold">
					Support as One <br />
					<span className="text-[#666666]">with the world's first DevCRM</span>
				</h2>
				<img
					src={ProductImage}
					alt="Product Image"
					className="absolute top-[20rem] max-[1300px]:left-[4rem] left-[8rem] w-[700px] h-auto"
				/>
			</section>
		</div>
	);
}
