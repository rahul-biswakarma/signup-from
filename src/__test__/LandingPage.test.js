import "@testing-library/jest-dom";
import { render, cleanup } from "@testing-library/react";

import LandingPage from "../components/Login";
import SignUpButtons from "../components/SignUpButtons";

afterEach(cleanup);

it("Landing Page snapshot check", () => {
	const { asFragment } = render(<LandingPage />);
	expect(asFragment(<LandingPage />)).toMatchSnapshot();
});

it("Continue of google buttons not disabled", () => {
	const { getByTestId } = render(<SignUpButtons />);
	expect(getByTestId("continue-with-google-button")).not.toHaveAttribute(
		"disabled"
	);
});

it("Continue of email buttons not disabled", () => {
	const { getByTestId } = render(<SignUpButtons />);
	expect(getByTestId("continue-with-email-button")).not.toHaveAttribute(
		"disabled"
	);
});
