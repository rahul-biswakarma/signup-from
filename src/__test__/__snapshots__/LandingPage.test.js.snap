// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page snapshot check 1`] = `
<DocumentFragment>
  <div
    class="w-full h-[100vh] overflow-hidden grid max-[800px]:grid-cols-[100%] max-[1200px]:grid-cols-[500px_auto] grid-cols-[700px_auto]"
    data-testid="login-component"
  >
    <section
      class="flex flex-col items-center justify-center bg-bg_1 border-r-[1px] border-border_color w-full h-full p-12"
    >
      <main
        class="max-w-[500px]"
      >
        <div>
          <a
            href="/"
          >
            <svg
              class="w-9 h-auto"
              fill="none"
              viewBox="0 0 44 44"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M42.7666 31.905a10.50012 10.50012 0 0 0-2.0142-5.6008 10.50171 10.50171 0 0 0-4.7348-3.607 10.5034 10.5034 0 0 0-5.935-.455c-1.9808.4238-3.7968 1.411-5.2292 2.8432a10.5026 10.5026 0 0 0-2.8446 5.2282 10.50364 10.50364 0 0 0 .454 5.9346 10.50313 10.50313 0 0 0 3.6064 4.735c1.6336 1.1978 3.5786 1.8974 5.601 2.0152a.58067.58067 0 0 0 .2296-.0378.5779.5779 0 0 0 .1962-.1252.57932.57932 0 0 0 .1308-.1926c.03-.0724.045-.15.0442-.2284v-9.9046h9.9116a.57767.57767 0 0 0 .2292-.045.58143.58143 0 0 0 .1928-.132.58042.58042 0 0 0 .125-.1972.5819.5819 0 0 0 .037-.2306ZM11.9791 42.1888 1.31526 24.0624a.57877.57877 0 0 1-.08135-.291.57915.57915 0 0 1 .07622-.2924.57758.57758 0 0 1 .21309-.2144.57824.57824 0 0 1 .29204-.078H23.1428c.1026 0 .2032.027.2918.0784a.58349.58349 0 0 1 .2132.214c.0508.089.0774.1896.077.2922a.58246.58246 0 0 1-.0798.2912L12.9812 42.1888a.58083.58083 0 0 1-.2126.2094.58164.58164 0 0 1-.2885.0766.58099.58099 0 0 1-.2884-.0766.57932.57932 0 0 1-.2126-.2094ZM22.5 10.9106a10.5009 10.5009 0 0 0-2.0152-5.60012 10.50341 10.50341 0 0 0-4.7351-3.60609 10.50519 10.50519 0 0 0-5.93484-.45413 10.50443 10.50443 0 0 0-5.2288 2.84362A10.50246 10.50246 0 0 0 2.19634 15.2565a10.50305 10.50305 0 0 0 3.60642 4.7347c1.63354 1.1976 3.57854 1.8972 5.60064 2.015a.58502.58502 0 0 0 .229-.039.57856.57856 0 0 0 .1955-.1254.58137.58137 0 0 0 .1309-.1918.5827.5827 0 0 0 .0455-.2278V11.503h9.9117a.5819.5819 0 0 0 .2258-.0447.58186.58186 0 0 0 .1908-.1287.58238.58238 0 0 0 .126-.1926.5843.5843 0 0 0 .0414-.2264Zm8.25 9.6466L20.084 2.4287a.58286.58286 0 0 1-.0798-.29134.58413.58413 0 0 1 .077-.29209.58288.58288 0 0 1 .2132-.21411.58167.58167 0 0 1 .2918-.07843h21.3276a.582.582 0 0 1 .5.29525.58365.58365 0 0 1 .0756.29094.58348.58348 0 0 1-.0798.28978L31.75 20.5572a.58092.58092 0 0 1-.2126.208c-.0876.05-.1866.076-.2874.076s-.2-.026-.2876-.076a.5819.5819 0 0 1-.2124-.208Z"
                fill="#000"
              />
            </svg>
          </a>
          <h1
            class="font-[500] text-md text-black mt-[0.5rem]"
          >
            Sign for DevRev
          </h1>
          <p
            class="text-black/40 text-md mt-[1rem]"
          >
            Welcome to DevRev, our mission is to help the world's most customer centric companies build and support great products.
          </p>
        </div>
        <form
          class="py-[3rem] flex flex-col gap-[2rem]"
        >
          <div>
            <label
              class=" text-black/60 text-sm"
              for="email"
            >
              Email
            </label>
            <div
              class="flex mt-[0.3rem]"
            >
              <span
                class="w-[3rem] h-[2.5rem] flex justify-center items-center bg-input_bg rounded-l"
              >
                <img
                  alt="id card icon"
                  class="w-5 h-auto"
                  src="email.svg"
                />
              </span>
              <input
                class="w-full h-[2.5rem] border border-input_bg rounded-r px-4 code-font focus:border-input_bg"
                data-testid="email-input"
                placeholder="name@work.com"
                required=""
                type="email"
              />
            </div>
          </div>
          <div>
            <label
              class=" text-black/60 text-sm"
              for="email"
            >
              Password
            </label>
            <div
              class="flex mt-[0.3rem]"
            >
              <span
                class="w-[3rem] h-[2.5rem] flex justify-center items-center bg-input_bg rounded-l"
              >
                <img
                  alt="id card icon"
                  class="w-5 h-auto"
                  src="lock.svg"
                />
              </span>
              <input
                class="w-full h-[2.5rem] border border-input_bg rounded-r px-4 code-font"
                data-testid="password-input"
                required=""
                type="password"
              />
            </div>
          </div>
          <button
            class="bg-blue_1 hover:bg-blue_2 p-[0rem_2rem] h-[2.5rem] rounded flex  items-center justify-center"
          >
            <span
              class="code-font text-sm text-white"
            >
              Login
            </span>
            <img
              alt="arrow icon"
              class="w-5 h-auto ml-[1rem]"
              src="arrow.svg"
            />
          </button>
        </form>
        <p
          class="text-sm text-black/40"
        >
          Don't have an account? 
          <span
            class="text-black font-semibold cursor-pointer"
          >
            <a
              href="/signup"
            >
              Sign Up
            </a>
          </span>
        </p>
      </main>
    </section>
    <section
      class="max-[800px]:hidden w-full h-full relative bg-loginHeroImage bg-center bg-cover no-repeat"
    />
  </div>
</DocumentFragment>
`;
